### PR TITLE
Add a note regarding consul_acl_token_secret_id in the consul_acl_token resource description

### PR DESCRIPTION
**Pull request by remilapeyre**
Also migrate consul_acl_token to tfplugindocs.

### CODE REVIEW

File: docs/resources/acl_token.md [hunk 1]
```diff
@@ -1,20 +1,22 @@
...
-# consul_acl_token
+# consul_acl_token (Resource)
 
 The `consul_acl_token` resource writes an ACL token into Consul.
 
-## Example Usage
+~> **NOTE:** The `consul_acl_token` resource does not save the secret ID of the generated token to the Terraform state to avoid leaking it when it is not needed. If you need to get the secret ID after creating the ACL token you can use the [`consul_acl_token_secret_id`](/docs/providers/consul/r/acl_token.html) datasource.
 
-### Basic usage
+## Example Usage
```
Comment (roncodingenthusiast) [inline docs/resources/acl_token.md:16]:
```suggestion
## Example usage
```
the docs team is moving towards sentence style capitalization for titles so maybe we could do the same here

File: docs/resources/acl_token.md [hunk 2]
```diff
@@ -26,72 +28,66 @@ resource "consul_acl_policy" "agent" {
...
+- `local` (Boolean) The flag to set the token local to the current datacenter.
+- `namespace` (String) The namespace to create the token within.
+- `node_identities` (Block List) The list of node identities that should be applied to the token. (see [below for nested schema](#nestedblock--node_identities))
+- `partition` (String) The partition the ACL token is associated with.
+- `policies` (Set of String) The list of policies attached to the token.
+- `roles` (Set of String) The list of roles attached to the token.
+- `service_identities` (Block List) The list of service identities that should be applied to the token. (see [below for nested schema](#nestedblock--service_identities))
 
-The `service_identities` block supports the following arguments:
+### Read-Only
```
Comment (roncodingenthusiast) [inline docs/resources/acl_token.md:62]:
```suggestion
### Read-only
```

File: docs/resources/acl_token.md [hunk 3]
```diff
@@ -26,72 +28,66 @@ resource "consul_acl_policy" "agent" {
...
 
-* `service_name` - (Required) The name of the service.
-* `datacenters` - (Required) The list of datacenters the policy is valid within.
+- `id` (String) The ID of this resource.
 
-The `node_identities` block supports the following arguments:
+<a id="nestedblock--node_identities"></a>
+### Nested Schema for `node_identities`
```
Comment (roncodingenthusiast) [inline docs/resources/acl_token.md:67]:
```suggestion
### Nested schema for `node_identities`
```

File: docs/resources/acl_token.md [hunk 4]
```diff
@@ -26,72 +28,66 @@ resource "consul_acl_policy" "agent" {
...
-* `description` - The description of the token.
-* `policies` - The list of policies attached to the token.
-* `roles` - The list of roles attached to the token.
-* `service_identities` - The list of service identities that should be applied to the token.
-* `node_identities` - The list of node identities that should be applied to the token.
-* `local` - The flag to set the token local to the current datacenter.
-* `expiration_time` - If set this represents the point after which a token should be considered revoked and is eligible for destruction.
-* `namespace` - The namespace to create the token within.
+<a id="nestedblock--service_identities"></a>
+### Nested Schema for `service_identities`
```
Comment (roncodingenthusiast) [inline docs/resources/acl_token.md:76]:
```suggestion
### Nested schema for `service_identities`
```

**Review by roncodingenthusiast — COMMENTED** [review]
This looks mostly good, just questions/comments on the docs

**Review by roncodingenthusiast — APPROVED** [review]
couple of docs nitpick and a question but otherwise this looks good to me. Approving so i dont block the progress of this PR

File: consul/resource_consul_acl_token.go [hunk 5]
```diff
@@ -65,9 +67,9 @@ func resourceConsulACLToken() *schema.Resource {
 							Description: "The name of the service.",
 						},
 						"datacenters": {
-							Type:        schema.TypeList,
-							Optional:    true,
-							Description: "Specifies the datacenters the effective policy is valid within.",
+							Type:     schema.TypeList,
+							Optional: true,
+							// Description: "The list of datacenters the policy is valid within.",
```
[the remark this answers is not in the document]
Comment (remilapeyre) [reply consul/resource_consul_acl_token.go:72]: No, thanks this was a mistake.

**Pull request merged**